File: moduletestcompat/src/main/java/com/munch/module/test/compat/MainActivity.java
package com.munch.module.test.compat;

import android.Manifest;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import com.munch.lib.log.LogLog;
import com.munch.lib.result.compat.PermissionListener;
import com.munch.lib.result.compat.PermissionLoopListener;
import com.munch.lib.result.compat.ResultHelperCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.tv).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ResultHelperCompat.requestPermission(MainActivity.this, 11, Manifest.permission.CALL_PHONE,
                        Manifest.permission.CAMERA, Manifest.permission.READ_CONTACTS)
                        .isLoop(true)
                        .loop(new PermissionLoopListener() {
                            @Override
                            public boolean loop(String permissions, int grantResults) {
                                LogLog.log(permissions, grantResults);
                                switch (permissions) {
                                    case Manifest.permission.CALL_PHONE:
                                        break;
                                    case Manifest.permission.CAMERA:
                                        return true;
                                    case Manifest.permission.READ_CONTACTS:
                                        break;
                                }
                                return false;
                            }
                        })
                        .result(new PermissionListener() {
                            @Override
                            public void result(String[] permissions, int[] grantResults) {
                                LogLog.log(permissions, grantResults);
                            }
                        });
            }
        });
    }
}

File: app/src/main/java/com/munch/project/app/MainActivity.kt
package com.munch.project.app

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.munch.lib.common.RouterHelper
import com.munch.lib.common.start2Component

/**
 * 此类作为壳引用启动页，主要用于xml注册，是Splash的占位
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        start2Component(RouterHelper.Test.MAIN)
        /*overridePendingTransition(R.anim.anim_mask_in, R.anim.anim_mask_in)*/
        finish()
    }
}